optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/LocalDatePicker.java

```diff
@@ -0,0 +1,49 @@
+package org.optaplanner.openshift.employeerostering.gwtui.client.common;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+
+import javax.inject.Inject;
+import javax.validation.ValidationException;
+
+import com.google.gwt.user.client.TakesValue;
+import elemental2.core.Date;
+import elemental2.dom.HTMLInputElement;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.optaplanner.openshift.employeerostering.shared.common.GwtJavaTimeWorkaroundUtil;
+
+@Templated
+public class LocalDatePicker implements TakesValue<LocalDate> {
+
+    @Inject
+    @DataField("date-picker")
+    private HTMLInputElement datePicker;
+
+    public boolean reportValidity() {
+        return datePicker.reportValidity();
+    }
+    
+    @Override
+    public void setValue(LocalDate value) {
+        datePicker.valueAsDate = new Date(Date.parse(value.toString()));
+    }
+
+    @Override
+    public LocalDate getValue() {
+        Date date;
+        if (reportValidity()) {
+            try {
+                date = datePicker.valueAsDate;
+            } catch (Exception e) {
```

Comment:
Can we catch anything less general then Exception?
It's usually a bad idea to catch Throwable, Error, Exception or RuntimeException, which a few notable exceptions. For example: any NullPointerException or OutOfMemoryError should bubble up to the default exception handler and show a big fat red error dialog, and not been seen as a "the user filled in an invalid date".

Reply:
Actually cannot test this case, the try/catch block is not checking for "the user filled in an invalid date?" (`reportValidity()` return true iff it is a valid date), but rather "does the browser support the `valueAsDate` computed field?" Both Chrome and Firefox support it, but I don't know if IE supports it.

Reply:
I understand, but we don't support IE, only Chrome, Firefox and probably Safari and Edge. Feel free to completely ignore Internet Explorer, it's end of life.

I've removed the Exception catches and tested on Firefox and Chrome successfully.

As for Edge, I 'd rather see proof that it doesn't work before introducing code that we might not need. QA will be certifying against multiple browsers, so they'll detect it if it's a problem. Alternatively, we should try to get Edge and Safari running on our system and simply try it, but that's challenging.